SCRetentionCompliancePolicy: No changes possible because of undeployed prev changes

ISSUE DESCRIPTION (this template):
When I create a new policy for the first time, the DSC resource works fine.
When I change the comment attribute of the resource manually (without DSC), I receive an error message that no changes can be made due to prev. pending changes.
However, even if I wait 2h and try manually the Set-ComplianceRetentionPolicy -RetryDistribution command, DSC does not run successfully.
![image](https://user-images.githubusercontent.com/88185985/209966021-4a180323-a154-405c-8fb3-df8a7dc778a1.png)

When I try to change the res manually in PS, I receive a similar error. 
![image](https://user-images.githubusercontent.com/88185985/209966170-bef2b7b0-bb66-488e-ab16-6483788da03a.png)

Solution for PS is to use the -force Parameter. When I looked in the DSC res Src code, the Set-ComplianceRetentionPolicy does not have a force parameter. Please add this parameter to the res.
![image](https://user-images.githubusercontent.com/88185985/209966231-201446c5-0047-424b-898f-2b1e47125d69.png)

#### Details of the scenario you tried and the problem that is occurring
Create res as described in issue ID 
Change something like the comment in the newly created policy.
Build the MOF and apply the config

Error in DSC Log:
##[error]Previous changes to the policy '1aefdbc5-5987' are being deployed. Once deployed, additional actions can be performed. Please refer to '[https://go.microsoft.com/fwlink/?linkid=2148779'](https://go.microsoft.com/fwlink/?linkid=2148779%27) to learn more.

![image](https://user-images.githubusercontent.com/88185985/210224898-d69cb68e-d8ab-41a1-8145-2cefd8af518d.png)


```powershell
Configuration MainConfig
{
    param
    (
         [Parameter(Mandatory = $true,Position=0)][ValidateNotNull()][System.Management.Automation.PSCredential]$AdminCred
    )
    Import-DscResource -ModuleName "Microsoft365DSC"
    Node localhost
    {
                
        SCRetentionCompliancePolicy 'RetentionCompliancePolicyEXO'
        {
                Name = "EXOPolicyNeverExisted3";
                Comment = "MyComment2";
                RestrictiveRetention = $False;
                ModernGroupLocation = "All"; #in 1012 version is empty, in latest version empty
                ExchangeLocation = "All"; #in 1012 emtpy, in latest version empty
                Ensure = "Present";
                Enabled = $False;
                Credential = $AdminCred;
        }
        SCRetentionCompliancePolicy 'RetentionCompliancePolicyTeams'
        {
                Name = "TeamsPolicyNeverExisted3";
                Comment = "MyComment2";
                RestrictiveRetention = $False;
                TeamsChatLocation = "All"; #in 1012 emtpy, in latest version empty
                Ensure = "Present";
                Enabled = $False;
                Credential = $AdminCred;
        }
    }
}
```

#### Suggested solution to the issue
Add -force Parameter to the DSC res and use this with the Set-ComplianceRetentionPolicy cmd in the Set method

The operating system the target node is running
OsName : Microsoft Windows Server 2019 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture : 64-Bit
WindowsVersion : 1809
WindowsBuildLabEx : 17763.1.amd64fre.rs5_release.180914-1434
OsLanguage : en-US
OsMuiLanguages : {en-US, de-DE}

Version of the DSC module that was used ('dev' if using current dev branch)
1.22.1221.1

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,9 @@
 * IntuneASRRulesPolicyWindows10
   * Corrects possible values for parameter OfficeCommunicationAppsLaunchChildProcess
     FIXES [#2730](https://github.com/microsoft/Microsoft365DSC/issues/2730)
+* SCRetentionCompliancePolicy
+  * Forces changes to existing policies to be applied.
+    FIXES [#2719](https://github.com/microsoft/Microsoft365DSC/issues/2719)
 * TeamsIPPhonePolicy
   * Added descriptions to the resource parameters
   * Limited possible parameter values where required

--- a/Modules/Microsoft365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.psm1
+++ b/Modules/Microsoft365DSC/DSCResources/MSFT_SCRetentionCompliancePolicy/MSFT_SCRetentionCompliancePolicy.psm1
@@ -768,7 +768,7 @@ function Set-TargetResource
         }
 
         Write-Verbose "Updating Policy with values: $(Convert-M365DscHashtableToString -Hashtable $CreationParams)"
-        Set-RetentionCompliancePolicy @CreationParams
+        Set-RetentionCompliancePolicy @CreationParams -Force
     }
     elseif (('Absent' -eq $Ensure) -and ('Present' -eq $CurrentPolicy.Ensure))
     {

